

Use iris flower dataset from sklearn library and use cross_val_score against following models to measure the performance of each. In the end figure out the model with best performance,

  1.  Logistic Regression
  2.  SVM
  3.  Decision Tree
  4.  Random Forest



In [2]:
from sklearn.datasets import load_iris
from sklearn.model_selection import cross_val_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
import numpy as np

In [4]:
iris = load_iris()

In [5]:
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

# 1. Logistic Regression

In [10]:
scores_logistic = cross_val_score(LogisticRegression(solver='lbfgs', max_iter=1000), iris.data, iris.target)
scores_logistic

array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

# 2. SVM

In [12]:
scores_svm = cross_val_score(SVC(), iris.data, iris.target)
scores_svm

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

# 3. Decision Tree

In [14]:
scores_dt = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
scores_dt

array([0.96666667, 0.96666667, 0.9       , 0.93333333, 1.        ])

# 4. Random Forest

In [16]:
scores_rf = cross_val_score(RandomForestClassifier(), iris.data, iris.target)
scores_rf

array([0.96666667, 0.96666667, 0.93333333, 0.93333333, 1.        ])

# Comparing averages of scores

In [21]:
scores_logistic.mean()

0.9733333333333334

In [22]:
scores_svm.mean()

0.9666666666666666

In [23]:
scores_dt.mean()

0.9533333333333334

In [24]:
scores_rf.mean()

0.96

So we conclude that the best score so far is Logistic Regression (0.97)